starknet_devnet/starknet_wrapper.py

```diff
@@ -89,45 +99,30 @@ async def deploy(self, contract_definition: ContractDefinition, contract_address
             constructor_calldata=constructor_calldata,
             contract_address_salt=contract_address_salt
         )
-        await self.update_state()
+        await self.__update_state()
 
-        self.address2contract_wrapper[contract.contract_address] = ContractWrapper(contract, contract_definition)
+        self.__address2contract_wrapper[contract.contract_address] = ContractWrapper(contract, contract_definition)
 
+    # pylint: disable=too-many-arguments
```

Comment:
Maybe the argument could be a single Object/Dataclass with all these arguments as properties